drake/util/TrigPoly.h

```diff
@@ -3,8 +3,30 @@
 
 #include "drake/util/Polynomial.h"
 #include <map>
-//#include <iostream>  // just for debugging
 
+/// A scalar multi-variate polynomial containing sines and cosines.
+/**
+ * TrigPoly represents a Polynomial some of whose variables actually represent
+ * the sines or cosines of other variables.  Sines and cosines of first-order
+ * polynomials (affine combinations of variables) are decomposed into
+ * polynomials of the sines and cosines of individual variables via the
+ * Prosthaphaeresis formulae.
```

Comment:
Can you clarify that?  testTrigPoly.cpp uses `(sin(p) * p * p + cos(p))` in the test and it seems to work, so mixing `p` and `sin(p)` is at least a tested case.